src/app/visit/visit.component.ts

```diff
@@ -161,6 +161,19 @@ export class VisitComponent implements OnInit, AfterViewInit, OnDestroy {
                 this.justIds = this.visitPoints.map((vp) => vp._id);
             } else {
                 this.visitPoints.push(row);
+                this.visitService
+                    .getBetterQualityOfImageFromStore(row.code)
+                    .subscribe({
+                        next: (data) => {
+                            const imageIndex = this.visitPoints.findIndex(
```

Comment:
why are you looking for index in `visitPoints` ? is it the place where you want to swap the images?

Reply:
just as a hint again! 

```
var x = [{name: 'a'}, {name: 'b'}, {name: 'c'}]
// to change name: 'c' to 'd' you need to:
var index = x.findIndex(obj => obj.name === 'c')
x[index].name = 'd'

// this will give you 
console.log(x) --> [{name: 'a'}, {name: 'b'}, {name: 'd'}]
```

apply the logic!!

Reply:
the data that we want to replace must have some sort of a filtered image.code that is found in visit points, so hence the search for the position of the image in visit points @macrog

Reply:
### `Check it out`:

![Uploading 2023-11-04_19-28-32_V1.gif…]()

Reply:
check it out on whatsapp

Reply:
but why `visitPoints` ? you don't understand the basics here. The table where you want to replace the image binds to `datasource` ❗ So you should be looking for the index in the `datasource` as well ❗ ❗